app/models/manageiq/providers/base_manager/refresher.rb

```diff
@@ -138,6 +140,31 @@ def save_inventory(ems, _target, persister)
         InventoryRefresh::SaveInventory.save_inventory(ems, persister.inventory_collections)
       end
 
+      def publish_inventory(ems, target, persister)
+        return unless publish_inventory?
+
+        messaging_client = MiqQueue.messaging_client('event_handler')
+        return if messaging_client.nil?
+
+        sender = "#{ems.emstype}__#{ems.name}"
+
+        persister.collections.each_value do |collection|
+          collection.to_hash[:data].to_a.each do |data|
+            # TODO pull the manager ref out of the hash data
+            target_identifier = collection.name
+
+            messaging_client.publish_topic(
```

Comment:
We're going to need a bulk produce option for ManageIQ::Messaging, publishing and waiting one message at a time is simply too slow.

The intended pattern for RdKafka is [[ref]](https://github.com/appsignal/rdkafka-ruby#producing-messages)
```
delivery_handles = data.each { |payload| producer.publish(:topic => topic, :payload => payload) }
delivery_handles.each(&:wait)
```

In local testing this cuts the time to produce messages down from 33.8s to 1.1s
```

Reply:
This has been merged to manageiq-messaging and released in v1.0.3